go/vt/mysqlctl/replication.go

```diff
@@ -197,12 +197,33 @@ func (mysqld *Mysqld) SetSuperReadOnly(on bool) error {
 
 // WaitMasterPos lets slaves wait to given replication position
 func (mysqld *Mysqld) WaitMasterPos(ctx context.Context, targetPos mysql.Position) error {
-	// Get a connection.
-	conn, err := getPoolReconnect(ctx, mysqld.dbaPool)
-	if err != nil {
-		return err
+	var conn mysql.ReplicationActor
+
+	// First check if filePos flavored Position was passed in. If so, we can't defer to the flavor in the connection,
+	// unless that flavor is also filePos.
+	if targetPos.MatchesFlavor(mysql.FilePosFlavorID) {
+		filePositioner := func(cp *mysql.ConnParams) *mysql.ConnParams {
```

Comment:
Instead of faking the connection flavor (which could have unintended side effects), could we just add methods `conn.MasterFilePosition()` and `conn.WaitUntilFilePositionCommand()`? Hopefully the methods don't even need to be flavor-specific, if all the flavors support file-based waits in the same way.

Reply:
Oh, I like that idea better actually. Ok, I'll refactor.

Reply:
+1

Reply:
Done. PTAL when you have a chance.